tests/sla_tests.js

```diff
@@ -82,10 +84,42 @@ var getMockClient = function() {
         cb(null, bridges[bridges.length-1]);
       }
     };
-    this.Channel= function() {
+    this.Channel = function() {
       var newChan = getMockChannel();
       return newChan;
     };
+    this.deviceStates = {
+      update: function(param, cb) {
+        var exists = mockDeviceStates.filter(function(deviceState) {
```

Comment:
'exists' implies this is a boolean. I would rename it to something else since it's used later in to establish whether or not the array is empty.

Another options is to do:

var exists = !mockDeviceStates.filter(function(deviceState) {
  return deviceState.deviceName === param.deviceName;
}).length;

but that is definitely harder to read.

The best option is to use a library like https://lodash.com to overcome some of JavaScript limitations. You could can do something like:

var exists = _.some(mockDeviceStates, function(deviceState) {
  return deviceState.deviceName === param.deviceName;
});